File: leetcode107.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
	vector<vector<int> > res;
	void helper(TreeNode *node, int height)
	{
		if(node == NULL)
			return;
		while(res.size() <= height)
			res.push_back(vector<int>());
		res[height].push_back(node->val);
		helper(node->left, height+1);
		helper(node->right, height+1);
	}

    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        helper(root,0);
        reverse(res.begin(), res.end());
        return res;
    }
};

File: leetcode107.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        helper(root,res,0);
        Collections.reverse(res);
        return res;
        
    }
    
    public void helper(TreeNode node, List<List<Integer>> res,int height)
    {
        if(node == null)
            return;
        if(res.size() <= height)
        {
            res.add(new ArrayList<Integer>());
        }
        res.get(height).add(node.val);
        //res.get(height).add(node.val);
        helper(node.left,res,height+1);
        helper(node.right,res,height+1);
    }
}
